src/extension/tools/node/getTaskOutputTool.tsx

```diff
@@ -30,16 +31,19 @@ export class GetTaskOutputTool implements vscode.LanguageModelTool<ITaskOptions>
 		@ITasksService private readonly tasksService: ITasksService,
 		@IWorkspaceService private readonly workspaceService: IWorkspaceService,
 		@IPromptPathRepresentationService private readonly promptPathRepresentationService: IPromptPathRepresentationService,
+		@ILogService private readonly logService: ILogService
 	) { }
 
 	async invoke(options: vscode.LanguageModelToolInvocationOptions<ITaskOptions>, token: vscode.CancellationToken) {
 		const label = this.getTaskDefinition(options.input)?.taskLabel;
 		if (!label) {
+			this.logService.logger.debug('getTaskOutputTool returning undefined: no matching label for task ' + options.input.id);
 			return;
 		}
 		// TODO:@meganrogge when there's API to determine if a terminal is a task, improve this vscode#234440
 		const terminal = this.terminalService.terminals.find(t => t.name === label);
 		if (!terminal) {
+			this.logService.logger.debug('getTaskOutputTool returning undefined: no terminal for task: ' + options.input.id + ' label: ' + label + ' terminal names: ' + this.terminalService.terminals.map(t => t.name).join(', '));
 			return;
 		}
 		const buffer = this.terminalService.getBufferForTerminal(terminal, Math.min(options.input.maxCharsToRetrieve ?? 16000, 16000));
```

Comment:
Probably down after "Output for task" we should say "Task is still running" etc. to reflect its state